core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_instruction/enclave_plan_instruction_impl.go

```diff
@@ -0,0 +1,57 @@
+package enclave_plan_instruction
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_capabilities"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+type EnclavePlanInstructionImpl struct {
+	// we do this way in order to have exported fields which can be marshalled
+	// and an unexported type for encapsulation
+	privateEnclavePlanInstruction *privateEnclavePlanInstruction
+}
+
+type privateEnclavePlanInstruction struct {
+	KurtosisInstructionStr string
+	Capabilities           *enclave_plan_capabilities.EnclavePlanCapabilities
```

Comment:
Any pros for splitting the instruction and the capability object here? We already have a ton of objects for pretty much anything, I don't feel too nervous about flattening the structure a bit here and having all the capabilities field at this level